sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -58,22 +89,26 @@
 
 import javax.annotation.Nullable;
 
+
 /**
  * Utility implementing the necessary reflection for working with {@link DoFnWithContext}s.
  */
 public abstract class DoFnReflector {
 
+  private static final String FN_DELEGATE_FIELD_NAME = "delegate";
+
   private interface ExtraContextInfo {
-    /**
-     * Create an instance of the given instance using the instance factory.
-     */
-    <InputT, OutputT> Object createInstance(
-        DoFnWithContext.ExtraContextFactory<InputT, OutputT> factory);
 
     /**
      * Create the type token for the given type, filling in the generics.
```

Comment:
Which given type? It takes two types and returns an unknown third. I can understand from reading the code below that an `ExtraContextInfo` represents a dynamically typed parameter in which `InputT` and `OutputT` may or may not occur. More explanation might help the next person to read this.